README.md

```diff
@@ -18,25 +20,185 @@ $ npm install --save pipage
+
+// Find a stream in the pipeline:
+var index = pipeline.indexOf( stream )
+var lastIndex = pipeline.lastIndexOf( stream )
+```
+
+### Example
+
+```js
+var Pipage = require('pipage')
+var path = require('path')
+var zlib = require('zlib')
+var unbzip2 = require('unbzip2-stream')
+var xz = require('xz')
+
+var pipeline = new Pipage()
+
+switch( path.extname( filename ) ) {
+  case '.gz':
+    pipeline.prepend( zlib.createUnzip() );
+    break
+  case '.bz':
+  case '.bz2':
+    pipeline.prepend( unbzip2() );
+    break
+  case '.xz':
+    pipeline.prepend( new xz.Decompressor() )
+    break
 }
 
-new BlockReadStream( '/path/to/resin-os.img' )
+fs.createReadStream( filename )
   .pipe( pipeline )
-  .pipe( new BlockWriteStream( '/dev/rdisk2' ) )
-  .once( 'finish', function() {
-    // happy face
-  })
+  .pipe( fs.createWriteStream( destination ) )
+```
+
+### Events
+
+#### Error handling
+
+Errors from streams within the pipeline are listened to,
+and re-emitted on the pipeline itself, with an additional `.stream` property
+being set on the `error` object, which is the stream that emitted it:
+
+```js
+pipeline.on( 'error', function( error ) {
+  // This error originated from the pipeline's internal stream
+  // available as `error.stream`, not from the pipeline itself
+  if( error.stream ) {
+    // ...
+  } else {
+    // This error came from the pipeline itself
+  }
+})
+```
+
+#### Binding to contained stream's events
+
+```js
+var stream = pipeline.get(-1)
+
+// Re-emit a stream's events on the pipeline:
+pipeline.bind( stream, 'eventname' )
+pipeline.bind( stream, [ 'someevent', 'otherevent' ])
+
+// Remove re-emission of a stream's event:
+pipeline.unbind( stream, 'eventname' )
+
+// Stop re-emission of all of stream's events on the pipeline:
+pipeline.unbindAll( stream )
+```
+
+##### Example
+
+```js
+var Pipage = require('pipage')
+
+// Let's say we have a stream which emits an event
+// we want to capture without having to get a reference to
+// that particular stream:
+module.exports = function createPipeline() {
+
+  var checksumStream = createChecksumStream( 'sha256', 'md5' )
+  var pipeline = new Pipage([ checksumStream ])
+
+  // This will cause the pipeline to re-emit
+  // the 'checksums' event from the `checksumStream`
+  pipeline.bind( checksumStream, 'checksums' )
+
+  // Add some more fancy things to the pipeline...
+
+  return pipeline
+
+}
+```
+
+```js
+var createPipeline = require('./create-pipeline')
+var pipeline = createPipeline()
+
+// Now we can listen for the bound event directly on the pipeline
+pipeline.on( 'checksums', function( checksums ) {
+  // Validate the checksums, etc...
+})
+
+fs.createReadStream( filename )
+  .pipe( pipeline )
+  .resume()
+```
+
+### Nested Pipelines
+
+Since pipelines are duplex streams, and contain duplex streams, they can be nested arbitrarily:
+
+```js
+var pipeline = new Pipage([
+  new Pipage(),
+  new Pipage([
+    new Pipage()
+  ])
+])
```

Comment:
Generally you'd almost never want to have the same stream instance in two places, as we'd already mentioned on another issue or pr, as that'd probably mess with the data flow & backpressure handling (unless the stream is designed to handle it).

Anyways, it depends very much on what that particular stream is doing, as the pipeline only cares about passing data from on stream to the next. So as long as it reads & writes data, nothing unexpected will happen.